src/clj/puppetlabs/puppetserver/certificate_authority.clj

```diff
@@ -1187,8 +1205,10 @@
                          #{:key-encipherment
                            :digital-signature} true)
                        (utils/subject-key-identifier
-                         subj-pub-key false)]]
-    (concat base-ext-list csr-ext-list)))
+                        subj-pub-key false)]
+        subject (get-csr-subject csr)
+        modded-csr-list (ensure-ext-list-has-cn-san subject csr-ext-list)]
```

Comment:
I would ideally like to pass `(concat csr-ext-list base-ext-list)` to `ensure-ext-list-has-cn-san`, but for reasons I don't quite understand, those two lists are `clojure.lang.PersistentVector`, but when `concat`'d, it becomes a `clojure.lang.LazySeq` and is incompatible with the function, generating the error:


```
Internal Server Error: java.lang.ClassCastException: clojure.lang.LazySeq cannot be cast to clojure.lang.Associative
```

Reply:
I think this is from our usage of `update-in` in `ensure-ext-list-has-cn-san` which isn't defined on a list, which is what concat returns.

I think alternative ways of doing it would be to call `vec` on the list, either before or calling or inside of `ensure-ext-list-has-cn-san`, call `(apply conj ...)` instead of `concat`, or reworking `ensure-ext-list-has-cn-san` to not use `update-in`.